### PR TITLE
Modify install script to enable SLE Micro rpm installation

### PR DESCRIPTION
**Pull request by galal-hussein**
#### Proposed Changes ####

- Enable RPM installation via zypper and transactional-update in SLE Micro distro, which will in turn enable support for Selinux support as well since the rpm installation installs the following packages
  - rke2-selinux
  - rke2-common
  - rke2-server/rke2-agent  
 

#### Verification ####

- run install script in sle micro
- reboot

The install script should install the rpm packages via transactional-updates and will require rebooting of the machine to take effect

#### Linked Issues ####

- https://github.com/rancher/rke2/issues/877

### CODE REVIEW

**Review by dweomer — APPROVED** [review]

File: install.sh [hunk 1]
```diff
@@ -404,18 +406,32 @@ install_airgap_tarball() {
...
-                maj_ver="7"
-                ;;
-        esac
+    . /etc/os-release
+    if [ -r /etc/redhat-release ] || [ -r /etc/centos-release ] || [ -r /etc/oracle-release ] || [ "${ID_LIKE%%[ ]*}" = "suse"  ]; then
+        repodir=/etc/yum.repos.d
+        if [ -d /etc/zypp/repos.d ]; then
+            repodir=/etc/zypp/repos.d
+        fi
+        if [ "${ID_LIKE%%[ ]*}" = "suse" ]; then
```
Comment (brandond) [inline install.sh:415]: The `"${ID_LIKE%%[ ]*}`  here looks like a bash-ism, does it work on busybox /bin/sh?

Comment (galal-hussein) [reply install.sh:415]: I am not sure, but I guess not, but do we support installation on busybox anyway?

Comment (briandowns) [reply install.sh:415]: I'd vote to make everything as POSIX compatible as possible.

Comment (brandond) [reply install.sh:415]: We need to support basic Bourne shell syntax since we run the installer with /bin/sh; this means no bash-isms.

Comment (galal-hussein) [reply install.sh:415]: makes sense, I will check for compatibility but FWIW we use the same syntax in k3s install script https://github.com/k3s-io/k3s/blob/master/install.sh#L463

Comment (brandond) [reply install.sh:415]: I was mostly concerned about the bracket expansion; if it works in K3s it should work here too!
```console
/ # ID_LIKE="suse foo bar"; echo ${ID_LIKE%%[ ]*}
suse
```

**Review by brandond — APPROVED** [review]

File: install.sh (hunk 1 above)
Comment (galal-hussein) [reply install.sh:415]: yeah I tested it and its  working as expected in busybox /bin/sh

**Review by briandowns — APPROVED** [review]

**Comment by galal-hussein** [comment]
BTW Smoke test for opensuse microos is failing because its using the stable channel which doesnt have the package right now, but should work with the testing channel, provided that the test will reboot the machine since its using transactional-update

**Pull request merged**